presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py

```diff
@@ -23,7 +27,7 @@ def __init__(self, analyzer_engine: AnalyzerEngine = None, ocr: OCR = None):
             ocr = TesseractOCR()
         self.ocr = ocr
 
-    def analyze(self, image: object, **kwargs) -> List[ImageRecognizerResult]:
+    def analyze(self, image: Optional[object], **kwargs) -> List[ImageRecognizerResult]:
```

Comment:
Thank you for writing out the potential alternatives.

The current approach is Approach 2 you described above, where `ocr_threshold` is removed from `**kwargs` prior to being passed into `AnalyzerEngine.analyze` (see lines 41-45).

If we want to go back to explicitly defining `ocr_threshold`, then I can revert this PR to [c35ffb1e86659c5bc9cd44fab70fc85c1ba34b3d](https://github.com/microsoft/presidio/tree/c35ffb1e86659c5bc9cd44fab70fc85c1ba34b3d)

Reply:
The downside of going back to the previous commit is that more files are changed, but if we feel that brings clarity, then it could be worth it.

Reply:
As discussed prior, the only place where `ocr_threshold` is really applied is directly in `ImageAnalyzerEngine.analyze`, which is called by `ImageAnalyzerEngine`, `ImagePiiVerifyEngine`, `DicomImageAnalyzerEngine`, and `DicomPiiVerifyEngine`. This is why this current version of the PR uses the `**kwargs` approach to reduce repeatition.

Reply:
And apologies I missed the last line of your original comment. I'll schedule some time for us to discuss. I agree it may be easier to chat in real-time about the pros and cons of the different approaches.

Reply:
@SharonHart - FYI, we have decided to go forward with an approach where OCR parameters are specified in the Optional argument `ocr_kwargs`, keeping it separate from the general `**kwargs` used by the text AnalyzerEngine.